torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -22,73 +22,93 @@ AccumulateGrad::AccumulateGrad(Variable variable_)
   add_input_metadata(variable);
 }
 
-auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
-  // XXX: this method is not thread-safe!
-  check_input_variables("AccumulateGrad", grads, 1, 0);
-
-  if (!grads[0].defined())
-    return {};
-  if (variable.grad_fn())
-    throw std::logic_error("leaf variable has been moved into the graph interior");
-  if (!variable.requires_grad())
-    return {};
+void AccumulateGrad::accumulateGradAndCallHooks(
+    const Variable& variable,
+    at::Tensor variable_grad,
+    const at::Tensor& new_grad,
+    size_t num_expected_refs,
+    bool has_post_hooks,
```

Comment:
nit: maybe this does not need to be a special parameter. We just want to add `1` to `num_expected_refs` right?

Reply:
Good point, let me fix this.